packages/commerce-sdk-react/src/components/ShopperExperience/Page/index.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright (c) 2023, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React, {useContext, useEffect, useState} from 'react'
+import {Page as PageType} from '../types'
+import {Region} from '../Region'
+
+type ComponentMap = {
+    [typeId: string]: React.ComponentType
+}
+type PageProps = {
+    page: PageType
+    components: ComponentMap
+    className?: string
+}
```

Comment:
@wjhsf I was trying to come up with a cleaner solution to this, maybe you can see whats wrong. 

Essentially I was trying to define the `PageProps` as an interface so I can use the extends syntax to extend the props/interface that a normal react html element can use. 

Like this:

```
interface PageProps extends React.ComponentPropsWithoutRef<'div'> {
    page: PageType
    components: ComponentMap
}
```

Later on in the component definition I would descruture the props to get all the props normally passable to a div like this:

```
const {page, components, ...rest} = props
```

And in the template portion of the component I could use the rest of those props like so:

```
<div id={id} {...rest}>
```

The advantage was that I don't have to declare `className` or `style` as props for the Page component, I jsut get them for free in its complete state.

But when I used the Page component I would get an error.

Reply:
The TS error:

```
Type '{ className: string; key: number; page: any; components: {}; }' is not assignable to type 'IntrinsicAttributes & PageProps'.
  Property 'className' does not exist on type 'IntrinsicAttributes & PageProps'.ts(2322)
  ```

Reply:
Figured it out.

Solution: 
```
interface PageProps extends React.ComponentProps<'div'> {
    page: PageType
    components: ComponentMap
}
```